### ISSUE TITLE
Graph UI Polish

### ISSUE THREAD

**Issue by axosoft-ramint** [gitkraken/vscode-gitlens#2120]
Issues related to user interaction with the graph that need to be addressed before release:

**Bugs:**

- [x] Changing repo (not by adding to workspace but by opening a different folder in VSCode) while the graph webview is open and focused causes it to close and the graph command disappears from the command palette (other GitLens commands are still there). **Solved.** Note: This was an issue with the VSIX using the Insiders version and getting overwritten by an auto-update.
- [x] The graph and its column headers appear misaligned initially when first loading, but are fixed by resizing the window (may be an issue on the component side, needs investigation).

**Tasks/Improvements:**

- [x] Top bar showing repo name and repo select dropdown should be a footer status bar instead (UX/visuals need discussion)
- [x] The dropdown to change repos within the graph webview should only be visible when you have multiple repos in the workspace
- [x] The graph should update when there are changes to the repo state (auto-fetch, new commits come in, etc.)
- [x] Double check the text styling in GitLens vs GK client and make sure it's similar (particularly for text in selected row)
- [x] Pass avatar urls for remote icons from GitLens when available based on provider (starting with GitHub)
- [ ] 💬 Hover interaction for commit rows (needs discussion)
- [x] 💬 Selection interaction for commit rows (needs discussion)
- [x] 💬 Click interaction for commit rows (needs discussion)
- [ ] 💬 Double-click interaction for commit rows (needs discussion)
- [x] 🛑 GitLens should pass custom icons into the graph for checked-out commit, stash, head, remote, tag (waiting on component-side update to allow icons to be passed in, needs discussion)
- [x] 🛑 GitLens should pass avatarUrl for graph rows (waiting on component-side update to allow avatar urls to be passed in, rather than using gravatar)
- [ ] 🛑 Send WIP information from GitLens to the graph to create a basic WIP node (waiting on component-side update to support basic WIP node)

**Comment by eamodio** [comment]
@axosoft-ramint Let's split out the work to be done here into a new issue to track for the next iteration. Thanks!

**Comment by axosoft-ramint** [comment]
Moving to a new issue for v2.
